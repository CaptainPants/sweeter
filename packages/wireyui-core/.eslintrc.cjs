module.exports = {
    extends: [
        '../../.eslintrc.cjs',
    ],
    parserOptions: {
        project: 'tsconfig.eslint.json',
        tsconfigRootDir: __dirname,
    },
    settings: {
        'import/resolver': {
            typescript: {
                project: ['./tsconfig.eslint.json'],
            }
        },
    },
    env: {
    },
    overrides: [],
    rules: {
    },
    plugins: [
    ],
};
