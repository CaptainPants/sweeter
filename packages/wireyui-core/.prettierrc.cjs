module.exports = require('../../common-config/prettierrc.cjs');