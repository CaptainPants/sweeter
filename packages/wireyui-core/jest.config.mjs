const common = require('../../common-config/jest.config.cjs');

module.exports = {
    ...common,
    globals: {
        'ts-jest': {
            tsConfig: 'tsconfig.withtests.json'
        }
    }
}