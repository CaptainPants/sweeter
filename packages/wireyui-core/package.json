{
    "name": "@captainpants/wireyui-core",
    "version": "0.0.3",
    "repository": "https://github.com/CaptainPants/wireyui-core",
    "author": "Stuart Murray <captainpantsnpm@gmail.com>",
    "license": "MIT",
    "private": false,
    "type": "module",
    "packageManager": "pnpm@8.6.7",
    "main": "./build/index.js",
    "types": "./build/index.d.ts",
    "exports": {
        ".": "./build/index.js",
        "./jsx-runtime": "./build/jsx-runtime.js"
    },
    "scripts": {
        "clean": "rimraf build && rimraf tsconfig.tsbuildinfo",
        "typecheck": "tsc",
        "format": "prettier --write src",
        "lint": "eslint src --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
        "test": "jest --silent=false src",
        "verify:fix": "corepack pnpm run format && corepack pnpm run lint --fix && corepack pnpm run test && corepack pnpm run typecheck",
        "verify": "corepack pnpm run lint && corepack pnpm run test && corepack pnpm run typecheck",
        "watch": "pnpm run copy_jsx_defs && tsc --watch",
        "build": "pnpm run copy_jsx_defs && tsc && rimraf --glob \"./build/**/*.test.{d.ts,js}\"",
        "copy_jsx_defs": "copyfiles -u 1 ./src/definitions.d.ts ./build"
    },
    "engines": {
        "pnpm": "8.6.7"
    }
}
