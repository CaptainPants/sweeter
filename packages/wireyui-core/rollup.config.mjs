export default {
    input: [
        './build/index.js',
        './build/jsx-runtime.js',
    ],
    output: {
        format: 'commonjs',
        dir: './build/cjs'
    },
  };