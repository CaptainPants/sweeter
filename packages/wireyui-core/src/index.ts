import './jsx-globals.js';

export * from './addStrongReference.js';
export * from './components/index.js';
export * from './context/index.js';
export * from './executionContext/index.js';
export * from './types.js';
export * from './signals/index.js';
export * from './flatten.js';
