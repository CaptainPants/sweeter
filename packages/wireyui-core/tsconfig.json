{
    "include": ["src/**/*.ts", "src/**/*.tsx"],
    "exclude": ["src/**/*.tests.ts", "src/**/*.tests.tsx"],
    "extends": "../../tsconfig.json",
    "compilerOptions": {
        "incremental": true,
        /* Emit */
        "outDir": "./build" /* Specify an output folder for all emitted files. */,
    }
}
