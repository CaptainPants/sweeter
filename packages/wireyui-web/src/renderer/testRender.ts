import { WebRenderer } from './WebRenderer.js';

export interface TestRenderResult {
    readonly nodes: NodeListOf<ChildNode>;
    readonly getHTML: () => string;
    readonly dispose: () => void;
}

/**
 * Convenience function for rendering isolated JSX chunks for unit tests
 * @param callback
 * @returns
 */
export function testRender(callback: () => JSX.Element): TestRenderResult {
    const renderer = new WebRenderer();

    const root = document.createElement('div');

    const dispose = renderer.start(root, callback);

    return {
        nodes: root.childNodes,
        getHTML: () => root.innerHTML,
        dispose,
    };
}
