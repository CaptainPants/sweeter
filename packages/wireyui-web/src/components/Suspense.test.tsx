import { Suspense } from "@captainpants/wireyui-core"
import { testRender } from "../renderer/testRender.js";

it('Suspense displays children', () => {
    const res = testRender(
        () => <Suspense fallback={() => <div>FALLBACK</div>}>{
            () => <div>COMPLETE</div>
        }</Suspense>
    );

    expect(res.getHTML()).toMatchSnapshot();
})