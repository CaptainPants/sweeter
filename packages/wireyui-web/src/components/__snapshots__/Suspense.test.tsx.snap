// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suspense displays children 1`] = `"<div>COMPLETE</div>"`;
